flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/utils/TableUtil.java

```diff
@@ -0,0 +1,432 @@
+	 * Find the indices of <code>targetCols</code> in string array <code>tableCols</code>. If
+	 * <code>targetCols</code> is
+	 * null, it will be replaced by the <code>tableCols</code>
+	 *
+	 * @param tableCols  a string array among which to find the targetCols.
+	 * @param targetCols the targetCols to find.
+	 * @return the indices of the targetCols.
+	 */
+	public static int[] findColIndices(String[] tableCols, String[] targetCols) {
+		if (targetCols == null) {
+			int[] indices = new int[tableCols.length];
+			for (int i = 0; i < tableCols.length; i++) {
+				indices[i] = i;
+			}
+			return indices;
+		}
+		int[] indices = new int[targetCols.length];
+		for (int i = 0; i < indices.length; i++) {
+			indices[i] = findColIndex(tableCols, targetCols[i]);
+		}
+		return indices;
+	}
+
+	/**
+	 * Find the index of <code>targetCol</code> from the <code>tableSchema</code>.
+	 *
+	 * @param tableSchema the TableSchema among which to find the targetCol.
+	 * @param targetCol   the targetCols to find.
+	 * @return the index of the targetCol.
+	 */
+	public static int findColIndex(TableSchema tableSchema, String targetCol) {
+		return findColIndex(tableSchema.getFieldNames(), targetCol);
+	}
+
+	/**
+	 * Find the indices of <code>targetCols</code> from the <code>tableSchema</code>.
+	 *
+	 * @param tableSchema the TableSchema among which to find the targetCols.
+	 * @param targetCols  the targetCols to find.
+	 * @return the indices of the targetCols.
+	 */
+	public static int[] findColIndices(TableSchema tableSchema, String[] targetCols) {
+		return findColIndices(tableSchema.getFieldNames(), targetCols);
+	}
+
+	/**
+	 * Find the types of the <code>targetCols</code>. If the targetCol not exist, return null.
+	 *
+	 * @param tableSchema TableSchema.
+	 * @param targetCols  the targetCols to find.
+	 * @return the corresponding types.
+	 */
+	public static TypeInformation[] findColTypes(TableSchema tableSchema, String[] targetCols) {
+		if (targetCols == null) {
+			return tableSchema.getFieldTypes();
+		}
+		TypeInformation[] types = new TypeInformation[targetCols.length];
+		for (int i = 0; i < types.length; i++) {
+			types[i] = findColType(tableSchema, targetCols[i]);
+		}
+		return types;
+	}
+
+	/**
+	 * Find the type of the <code>targetCol</code>. If the targetCol not exist, return null.
+	 *
+	 * @param tableSchema TableSchema
+	 * @param targetCol   the targetCol to find.
+	 * @return the corresponding type.
+	 */
+	public static TypeInformation findColType(TableSchema tableSchema, String targetCol) {
+		int index = findColIndex(tableSchema.getFieldNames(), targetCol);
+
+		return index == -1 ? null : tableSchema.getFieldTypes()[index];
+	}
+
+	/**
+	 * Determine whether it is number type, number type includes double, long, byte, int, float and short.
+	 *
+	 * @param dataType the dataType to determine.
+	 * @return whether it is number type
+	 */
+	public static boolean isNumber(TypeInformation dataType) {
+		return Types.DOUBLE == dataType
```

Comment:
what about: `Types.BIG_DEC` & `Types.BIG_INT`? they seems to be SQL only but should it be considered?

Reply:
Thanks, Types.BIG_DEC and Types.BIG_INT are not necessary for ML algorithms, so we change the function name to isSupportedNumericType